crypto/cmp/cmp_ctx.c

```diff
@@ -123,9 +128,16 @@ OSSL_CMP_CTX *OSSL_CMP_CTX_new(void)
 
  err:
     OSSL_CMP_CTX_free(ctx);
+    X509err(0, ERR_R_MALLOC_FAILURE);
     return NULL;
 }
 
+OSSL_CMP_CTX *OSSL_CMP_CTX_new(void)
```

Comment:
Do we need this variant. I'd be tempted to just have the one above and call *that* one `OSSL_CMP_CTX_new`. We couldn't do that with other APIs because of backwards compatibility - but that isn't an issue with CMP since its never appeared in a stable release.

Reply:
All right, done.
I've added the two new parameters also to `OSSL_CMP_SRV_CTX_new()`.

Reply:
@slontis, therefore I also had to adapt the invocations of these `new()` functions in the tests.
For now I've just put the defaults there: `(NULL, NULL)`.